core/src/main/java/org/apache/carbondata/core/util/BlockletDataMapUtil.java

```diff
@@ -264,6 +264,16 @@ public static boolean isSameColumnAndDifferentDatatypeInSchema(
             .equalsIgnoreCase(tableColumnList.get(i).getColumnName()) && !indexFileColumnList.get(j)
             .getDataType().getName()
             .equalsIgnoreCase(tableColumnList.get(i).getDataType().getName())) {
+          if ("varchar".equalsIgnoreCase(indexFileColumnList.get(j).getDataType().getName()) &&
+              "string".equalsIgnoreCase(tableColumnList.get(i).getDataType().getName())) {
+            throw new RuntimeException("Datatype of the Column "
```

Comment:
Please use IOException & add throws declaration to method so that callers are clear on how to handle it.
Or
I think all the callers are throwing exception. based on the flag. Instead you can return the custom message and throw Exception in caller if the msg is not empty.

Reply:
vishal ask don't use IOException.

Reply:
OK.. You can use InvalidConfigurationException

Reply:
org.apache.hadoop.mapreduce.RecordReader#initialize only support throw IOException, InterruptedException;, so we should convert InvalidConfigurationException 
 to IOException or InterruptedException in org.apache.carbondata.hadoop.CarbonRecordReader#initialize, org.apache.carbondata.hadoop.util.CarbonVectorizedRecordReader#initialize

is it ok?

Reply:
add org.apache.carbondata.hive.CarbonHiveRecordReader#initialize

Reply:
> org.apache.hadoop.mapreduce.RecordReader#initialize only support throw IOException, InterruptedException;, so we should convert InvalidConfigurationException
> to IOException or InterruptedException in org.apache.carbondata.hadoop.CarbonRecordReader#initialize, org.apache.carbondata.hadoop.util.CarbonVectorizedRecordReader#initialize
> 
> is it ok?

This is what I meant from my comment. "add throws declaration to method so that callers are clear on how to handle it."